### ISSUE TITLE
[PHP] Intermittent conformance failure with new C extension

### ISSUE THREAD

**Issue by haberman** [protocolbuffers/protobuf#7732]
The following error has been showing up intermittently on Kokoro. So far it has only appeared on Mac:

```
CONFORMANCE TEST BEGIN ====================================

ERROR, test=Required.Proto3.JsonInput.FloatFieldQuotedValue.JsonOutput: Output was not equivalent to reference message: modified: optional_float: 1 -> 113
. request=json_payload: "{\"optionalFloat\": \"1\"}" requested_output_format: JSON message_type: "protobuf_test_messages.proto3.TestAllTypesProto3" test_category: JSON_TEST, response=json_payload: "{\"optionalFloat\":113}"

These tests failed.  If they can't be fixed right now, you can add them to the failure list so the overall suite can succeed.  Add them to the failure list by running:
  ./update_failure_list.py failure_list_php_c.txt --add failing_tests.txt

  Required.Proto3.JsonInput.FloatFieldQuotedValue.JsonOutput
```

Another instance.

```
CONFORMANCE TEST BEGIN ====================================

ERROR, test=Required.Proto3.JsonInput.FloatFieldQuotedValue.ProtobufOutput: Output was not equivalent to reference message: modified: optional_float: 1 -> 11
. request=json_payload: "{\"optionalFloat\": \"1\"}" requested_output_format: PROTOBUF message_type: "protobuf_test_messages.proto3.TestAllTypesProto3" test_category: JSON_TEST, response=protobuf_payload: "]\000\0000A"
ERROR, test=Required.Proto3.JsonInput.FloatFieldQuotedValue.JsonOutput: Output was not equivalent to reference message: modified: optional_float: 1 -> 11
. request=json_payload: "{\"optionalFloat\": \"1\"}" requested_output_format: JSON message_type: "protobuf_test_messages.proto3.TestAllTypesProto3" test_category: JSON_TEST, response=json_payload: "{\"optionalFloat\":11}"

These tests failed.  If they can't be fixed right now, you can add them to the failure list so the overall suite can succeed.  Add them to the failure list by running:
  ./update_failure_list.py failure_list_php_c.txt --add failing_tests.txt

  Required.Proto3.JsonInput.FloatFieldQuotedValue.JsonOutput
  Required.Proto3.JsonInput.FloatFieldQuotedValue.ProtobufOutput
```

**Comment by haberman** [comment]
This appears to be a bug in the upb JSON parser when a float field is encoded inside a string.

**Comment by haberman** [comment]
This should be fixed by #7734.
